festim/hydrogen_transport_problem.py

```diff
@@ -90,24 +90,24 @@ class HydrogenTransportProblem:
     def __init__(
         self,
         mesh=None,
-        subdomains=[],
-        species=[],
-        reactions=[],
+        subdomains=None,
+        species=None,
+        reactions=None,
         temperature=None,
-        sources=[],
-        boundary_conditions=[],
+        sources=None,
+        boundary_conditions=None,
         settings=None,
-        exports=[],
-    ) -> None:
+        exports=None,
+    ):
         self.mesh = mesh
-        self.subdomains = subdomains
-        self.species = species
-        self.reactions = reactions
+        self.subdomains = subdomains or []
+        self.species = species or []
+        self.reactions = reactions or []
         self.temperature = temperature
-        self.sources = sources
-        self.boundary_conditions = boundary_conditions
+        self.sources = sources or []
+        self.boundary_conditions = boundary_conditions or []
         self.settings = settings
-        self.exports = exports
+        self.exports = exports or []
```

Comment:
Nice hack, I didn't know of this way of doing things.
Does it work for all python versions?

Also, we may want to add a comment to say what this does for people like me who don't know this